[yeobohshin] iP

YeoBohShin opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

YeoBohShin committed:
Rename, Greet, Exit

YeoBohShin committed:
Echo

YeoBohShin committed:
Add, List

YeoBohShin committed:
Mark as Done

YeoBohShin committed:
ToDos, Events, Deadlines

YeoBohShin committed:
Automated Text UI Testing

YeoBohShin committed:
Handle Errors

YeoBohShin committed:
Delete

YeoBohShin committed:
Added documentations

YeoBohShin committed:
Save

YeoBohShin committed:
Banish data.txt

YeoBohShin committed:
Add dates

YeoBohShin committed:
Use More OOP

YeoBohShin committed:
Organize into Package

YeoBohShin committed:
Merge branch 'add-gradle-support'

YeoBohShin committed:
Use Gradle

YeoBohShin committed:
Add JUnit Tests

YeoBohShin committed:
Create a JAR File

YeoBohShin committed:
Add Find

YeoBohShin committed:
Add javadocs

YeoBohShin committed:
Follow codingstandards

YeoBohShin committed:
Merge branch 'branch-javadocs'

YeoBohShin committed:
Merge branch 'branch-level-9'

YeoBohShin committed:
Add javadoc

Chandan8186 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,110 @@
+        }
+        return true;
+    }
+}
```
Perhaps the functionality under each case could be abstracted out into a separate function to improve readability?

Chandan8186 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,113 @@
+            throw new IllegalArgumentException("Oops! Something went wrong!");
+        }
+    }
+
```
Perhaps method updateStorage would be more suitable in Storage.java? Maybe it could accept the task list as a parameter?

Chandan8186 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,113 @@
+        System.out.println("Here are the matching tasks in your list:");
+        this.list.stream().filter(task -> task.description.contains(filterWord)).forEach(System.out::println);
+    }
+}
```
I like the use of streams to make finding easier 👍

Chandan8186 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+    private TaskList list;
+    private Ui ui;
+    private boolean isActive = true;
+    private final String FILE_PATH;
```
Perhaps there is no need for storing file path in Duke class since it is stored in Storage class?

Chandan8186 reviewed (commented):
Overall, there is mostly good use of abstraction/encapsulation. Other than the parse function in Parse class, your code seems easy to read and understand.

maypfv commented on src/main/java/duke/DateParser.java:
```diff
@@ -0,0 +1,24 @@
+package duke;
```
Perhaps you can package this class into a parser package in Duke. 
```suggestion
package duke.parser;
```

maypfv commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public Deadline(String description, boolean isDone, LocalDate deadline) {
+        super(description, isDone);
+        this.deadline= deadline;
```
Perhaps you can add a space between this.deadline and the = sign.

maypfv commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+package duke;
+import java.util.Scanner;
```
Maybe you can add a line between package and import statement.

maypfv commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        System.out.println(task);
+    }
+
+}
```
I like the names of the methods being easy to understand when used. 👍

maypfv reviewed (commented):
Looks good to me 👍 
Only some minor changes suggested. Overall, could do more packaging to organize.

YeoBohShin committed:
Add checkstyle

YeoBohShin committed:
Add GUI

YeoBohShin committed:
Merge branch 'branch-GUI'

brandon-nam commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+        String command = input[0];
+        String specifications = input.length > 1 ? input[1] : "";
+        switch (command) {
+        case "bye":
```
I like how the indentations of the switch-case is correct :)

brandon-nam commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+package duke;
```
Perhaps you can put Parser in a different package inside the duke package, like duke.parser?

brandon-nam commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+public class TaskList {
+
+    /** List to store tasks */
+    private List<Task> list;
```
Perhaps you could add task in the variable name to make it more identifiable like taskList or tasks?

brandon-nam reviewed (commented):
The coding standard looks good to me overall!

brandon-nam commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,61 @@
+package duke;
```
Perhaps different types of tasks could be packaged together and named duke.tasks?

YeoBohShin committed:
Add assertions

Assertions are needed to maintain the correctness of the code.

Assertions added to check if index of task is valid before adjusting the task list.

YeoBohShin committed:
Merge pull request #1 from YeoBohShin/branch-assertions

Add assertions

YeoBohShin committed:
Improve Code Quality

Made the code more cleaner and more readable.

Breakdown methods into different level of abstactions for better understanding

YeoBohShin committed:
Merge pull request #2 from YeoBohShin/branch-codequality

Improve Code Quality

YeoBohShin committed:
Add extensions

YeoBohShin committed:
Merge pull request #3 from YeoBohShin/branch-update

Add extensions

YeoBohShin committed:
Conform to codequality

YeoBohShin committed:
Fix bugs regarding update function

YeoBohShin committed:
Follows javadocs coding standards

YeoBohShin committed:
Adds testing for tasklist and updates find function

YeoBohShin committed:
Update User Guide

YeoBohShin committed:
Merge pull request #5 from YeoBohShin/branch-userguide

Update User Guide

YeoBohShin committed:
Update Ui.png

YeoBohShin committed:
Change bot greetings

YeoBohShin committed:
Remove unnecessary print line statements

YeoBohShin committed:
Update README.md and fix some bugs

YeoBohShin committed:
Fix bugs in update function for todo

YeoBohShin committed:
Update README.md

YeoBohShin committed:
Fix bugs for adding event task

YeoBohShin committed:
Update dialog box wrapping logic

YeoBohShin committed:
Fix README.md

YeoBohShin committed:
Update README.md
